src/server/server_dispatch.c

```
@@ -808,19 +808,78 @@ void bc_torpedo_hit_callback(int shooter_slot, i32 target_id,
+}
+
+/* Handle inbound owner StateUpdate (0x1C) as server input only.
+ * Downstream 0x1C stream is generated server-side in main.c. */
+static void handle_state_update_input(int peer_slot, bc_peer_t *peer,
+                                      const u8 *payload, int payload_len)
+{
+    if (g_registry_loaded && peer->has_ship) {
+        const bc_ship_class_t *cls =
+            bc_registry_get_ship(&g_registry, peer->class_index);
+        bc_state_update_t su;
+        if (bc_parse_state_update(payload, payload_len, &su)) {
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)